src/unstable-temp/DownloadImageModal/ImageForm.tsx

```diff
@@ -152,23 +157,42 @@ export const ImageForm = ({
 		);
 	}, [deviceType?.slug, rawVersion]);
 
+	const download = async (url: string) => {
+		try {
+			const res = await fetch(url, {
+				method: 'GET',
+				headers: new Headers({
+					Authorization: 'Bearer ' + authToken,
+				}),
+			});
+			const blob = await res.blob();
+			const newBlob = new Blob([blob]);
+
+			const blobUrl = window.URL.createObjectURL(newBlob);
```

Comment:
Gave a try on the dashboard and
```
<a href="https://api.balena-cloud.com/download?deviceType=raspberrypi3&version=2.83.21+rev1.prod" download="asdf.img">asdf</a>
```
(note the missing `target="_blank"`) opens the download prompt right away and defaults to the name that the API returns (`balena-cloud-raspberrypi3-2.83.21+rev1-v12.10.3.img`).
![Peek 2021-09-28 14-05](https://user-images.githubusercontent.com/1295829/135075616-2698aeb4-38f4-48af-b0c7-24d69a7ce6f6.gif)